tests/ts_setup.erl

```diff
@@ -0,0 +1,100 @@
+%% -*- Mode: Erlang -*-
+%% -------------------------------------------------------------------
+%%
+%% Copyright (c) 2015, 2016 Basho Technologies, Inc.
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+%% -------------------------------------------------------------------
+%% @doc A util module for riak_ts tests
+
+-module(ts_setup).
+
+-include_lib("eunit/include/eunit.hrl").
+
+-export([start_cluster/0, start_cluster/1, start_cluster/2,
+         conn/1, conn/2, stop_a_node/1, stop_a_node/2,
+         create_bucket_type/3, create_bucket_type/4,
+         activate_bucket_type/2, activate_bucket_type/3]).
+
+-spec start_cluster() -> list(node()).
+start_cluster() ->
+    start_cluster(1).
+
+-spec start_cluster(NumNodes :: pos_integer()) -> list(node()).
+start_cluster(NumNodes) ->
+    start_cluster(NumNodes, []).
+
+-spec start_cluster(NumNodes :: pos_integer(),
+                    Config :: list(tuple())) -> list(node()).
+start_cluster(NumNodes, Config) ->
+    rt:set_backend(eleveldb),
+    lager:info("Building cluster of ~p~n", [NumNodes]),
+    rt:build_cluster(NumNodes, Config).
+
+-spec conn(list(node())) -> pid().
+conn(Nodes) ->
+    conn(1, Nodes).
+
+-spec conn(pos_integer(), list(node())) -> pid().
+conn(Which, Nodes) ->
+    Conn = rt:pbc(lists:nth(Which, Nodes)),
+    ?assert(is_pid(Conn)),
+    Conn.
+
+-spec stop_a_node(list(node())) -> ok.
+stop_a_node(Nodes) ->
+    stop_a_node(2, Nodes).
+
+-spec stop_a_node(pos_integer(), list(node())) -> ok.
+stop_a_node(Which, Nodes) ->
+    ok = rt:stop(lists:nth(Which, Nodes)).
+
+-spec create_bucket_type(list(node()), string(), string()) -> ok.
+create_bucket_type(Nodes, DDL, Table) ->
```

Comment:
The two-step procedure to create a TS table, involving riak-admin, dates back to the time when we didn't have a working CREATE TABLE query. While it can be seen as making sense (if you squint) to, for instance, check that `riak-admin`'s function remains not broken, creating a table with a single query is a cleaner and perhaps faster means to our testing ends. I would suggest to ~~deprecate~~ delete the `create_bucket_type` and `activate_bucket_type` functions and replace them at call sites with `{ok, {[], []}} = riakc_ts:query(Client, DDL)`.